src/Collection.ts

```diff
@@ -26,14 +27,14 @@ export default class Collection extends Base {
   /**
    * Alias for models.map
    */
-  map (callback: (model: Model) => mixed): Array<*> {
+  map (callback: (model: T) => any): Array<any> {
```

Comment:
What about `map (callback: (model: T) => P): Array<P>` to ensure consitency between the return type of the map call back and the returned array.

Reply:
Hmm, I'm not sure a generic return type is the right approach. This implies that you should declare the return type of this callback function in the Collection instance declaration.

Reply:
mmm, I'm not sure either, but maybe something like this?

```typescript
map<P> (callback: (model: T) => P): Array<P>
```

so later you can do

```typescript
myCollection.map<string>((user) => user.name)
```

Reply:
Not even sure it wourld work xD

Reply:
However in my experience explicitly typing something like this would be more beneficial than problematic (except for the verbosity it would bring)

Reply:
🎉 you nailed it 🎉